python/dolfinx/fem/petsc.py

```diff
@@ -753,6 +747,8 @@ class LinearProblem:
     using PETSc as a linear algebra backend.
     """
 
+    _prefix_counter = 0
```

Comment:
What happens if `LinearProblem`s are created on some ranks and not others? Looks like the prefix will get messed up.

Reply:
Of the three options proposed (sharing state on comm, explicit passing prefix by user, a counter) the prefix_counter and different handling whether we are on COMM_WORLD etc. is my least favourite.

Reply:
I now tried with a new approach that mixes input argument and default internal construction

Reply:
Marking this as resolved in order to clean up the page a bit, you'll receive another message shortly after with the latest attempt ("forcing" users to provide a prefix)